src/main/java/com/github/movimentocodar/maquinadecafe/maquinadecafe/creditos/GerenciadorCredito.java

```diff
@@ -0,0 +1,27 @@
+package com.github.movimentocodar.maquinadecafe.maquinadecafe.creditos;
+
+import java.math.BigDecimal;
+
+public class GerenciadorCredito {
+    private BigDecimal creditos = BigDecimal.ZERO;
+
+    public void adicionarCredito(BigDecimal valorCredito, FormaDePagamento formaDePagamento) {
+        if (!valorIsValid(valorCredito) || !formaDePagamento.isValid(valorCredito))
+            throw new IllegalArgumentException("valorCredito não pode ser <=0");
+        creditos = creditos.add(valorCredito);
+    }
+
+    public void sacarCredito(BigDecimal valorCredito) {
+        if (!valorIsValid(valorCredito))
+            throw new IllegalArgumentException("valorCredito não pode ser <=0");
+        creditos = creditos.subtract(valorCredito);
+    }
+
+    public BigDecimal getCreditos() {
+        return creditos;
+    }
+
+    private boolean valorIsValid(BigDecimal valorCredito){
```

Comment:
um padrão de nomenclatura para métodos que retornam boolean é começar com `is`, poderia usar `isValorValido` nesse caso